### PR TITLE
Implement support for GHC-7.10 and GHC-8.0

### PR DESCRIPTION
**Pull request by lehins**
Backwards compatibility works. doctests had to be disabled for ghc-7.10

### CODE REVIEW

File: test/Spec.hs [hunk 1]
```diff
@@ -31,49 +30,51 @@ main =
...
+    , integralSpec (Proxy :: Proxy CUIntPtr)
+    , integralSpec (Proxy :: Proxy CIntMax)
+    , integralSpec (Proxy :: Proxy CUIntMax)
+    , integralSpec (Proxy :: Proxy Integer)
+    , integralSpec (Proxy :: Proxy Natural)
+    -- , bitmaskSpec (Proxy :: Proxy Word8)
+    -- , bitmaskSpec (Proxy :: Proxy Word16)
+    -- , bitmaskSpec (Proxy :: Proxy Word32)
+    -- , bitmaskSpec (Proxy :: Proxy Word64)
+    -- , bitmaskSpec (Proxy :: Proxy Word)
```
Comment (curiousleo) [inline test/Spec.hs]:
```suggestion
```

The definition of `bitmaskSpec` isn't even in the codebase anymore. Time to remove this.

File: random.cabal [hunk 2]
```diff
@@ -63,14 +63,26 @@ build-type:         Custom
 extra-source-files:
     README.md
     CHANGELOG.md
+tested-with:         GHC == 7.10.2
+                   , GHC == 7.10.3
+                   , GHC == 8.0.2
+                   , GHC == 8.2.2
+                   , GHC == 8.4.3
```
Comment (curiousleo) [inline random.cabal:70]:
```suggestion
```

8.4.3 is not enabled in `.travis.yml`.

File: random.cabal [hunk 3]
```diff
@@ -63,14 +63,26 @@ build-type:         Custom
...
+                   , GHC == 8.4.4
+                   , GHC == 8.6.3
+                   , GHC == 8.6.4
```
Comment (curiousleo) [inline random.cabal:73]:
```suggestion
```

8.6.3 and 8.6.4 are not enabled in `.travis.yml`.

File: random.cabal [hunk 4]
```diff
@@ -63,14 +63,26 @@ build-type:         Custom
...
+                   , GHC == 8.6.5
+                   , GHC == 8.8.1
+                   , GHC == 8.8.2
```
Comment (curiousleo) [inline random.cabal:76]:
```suggestion
                   , GHC == 8.8.3
```

File: random.cabal [hunk 5]
```diff
@@ -83,14 +95,17 @@ library
     hs-source-dirs:   src
     default-language: Haskell2010
     ghc-options:
-        -Wall -Wincomplete-record-updates -Wincomplete-uni-patterns
```
Comment (curiousleo) [inline random.cabal:86]: Are these warning not available in the old GHC version? I think they are valuable. If they are only available in newer versions, let's activate them in newer versions.

.. or alternatively, leave them as they are here and overwrite `ghc-options` in the existing `if impl(ghc < 8.0)` block below.

File: src/System/Random/Internal.hs [hunk 6]
```diff
@@ -415,8 +437,7 @@ runStateGenST g action = runST $ runStateGenT g action
 
 -- | The standard pseudo-random number generator.
 newtype StdGen = StdGen { unStdGen :: SM.SMGen }
-  deriving newtype (RandomGen, NFData)
-  deriving stock   (Show)
+  deriving (Show, RandomGen, NFData)
```
Comment (curiousleo) [inline src/System/Random/Internal.hs:440]: Haha, a battle of backwards vs. forwards compatibility. Fine, my understanding is that right now `DerivingStrategies` is more of a nice to have than a necessity.

File: random.cabal (hunk 2 above)
Comment (lehins) [reply random.cabal:70]: It should be, which means that is not the correct change.

File: random.cabal (hunk 3 above)
Comment (lehins) [reply random.cabal:73]: same here. It is not tested now, but it will be soon.

File: random.cabal (hunk 5 above)
Comment (lehins) [reply random.cabal:86]: I had a feeling you are going to say this :smile: That's right, they are not available on older ghc. I am usually all for theses warnings, but we aren't using records or pattern matches that might fail. Regardless of that fact, I'll add them back with a condition.

File: src/System/Random/Internal.hs (hunk 6 above)
Comment (lehins) [reply src/System/Random/Internal.hs:440]: Yep, precisely. :smile:

File: random.cabal (hunk 2 above)
Comment (curiousleo) [reply random.cabal:70]: Ah right, thanks. I guess once this PR and https://github.com/idontgetoutmuch/random/pull/148/files#diff-bb794c45fcbccfb6ea3e4c9ca0122833 are in, the `tested-with` list and the `.travis.yml` will be in sync.

**Pull request merged**